osaca/semantics/kernel_dg.py

```diff
@@ -48,6 +55,24 @@ def _extend_path(cls, dst_list, kernel, dg, offset):
             dst_list.extend(tmp_list)
         # print('Thread [{}-{}] done'.format(kernel[0]['line_number'], kernel[-1]['line_number']))
 
+    @staticmethod
+    def get_load_line_number(line_number):
+        # The line number of the load must be less than the line number of the instruction.  The
+        # offset is irrelevant, but it must be a machine number to avoid silly rounding issues.
```

Comment:
The literal `0.1` represents a machine number too (0.1000000000000000055511151231257827021181583404541015625). I guess you want it to be a machine number with lots of 0s, so that no rounding happens when it is added to small integers?